code_samples/catalog/custom_catalog_filter/src/CatalogFilter/ProductNameFilter.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\CatalogFilter;
+
+use Ibexa\Contracts\ProductCatalog\CatalogFilters\FilterDefinitionInterface;
+use Ibexa\Contracts\ProductCatalog\Values\Product\Query\Criterion\ProductName;
+use Ibexa\Contracts\ProductCatalog\Values\Product\Query\CriterionInterface;
+
+final class ProductNameFilter implements FilterDefinitionInterface
+{
+    public function getIdentifier(): string
+    {
+        return 'product_name';
+    }
+
+    public function getName(): string
+    {
+        return 'Product name';
+    }
+
+    public function supports(CriterionInterface $criterion): bool
+    {
+        return $criterion instanceof ProductName;
+    }
+
+    public static function getTranslationMessages(): array
+    {
+        return [];
+    }
+
+    public function getGroupName(): string
```

Comment:
If this is relevant, to translate the Group name it should be 
```
    public function getGroupName(): string
    {
        return $this->translator->trans(
            /** @Desc("Custom filters") */
            'filter.group.custom.label',
            [],
            'product_catalog'
        );
    }
```

Reply:
To simplify the example (which doesn't relate to translations), let's keep this omitted.